utils/hooks/useFetchData.js

```diff
@@ -0,0 +1,31 @@
+import { useState, useEffect } from "react";
+import { captureException } from "@sentry/nextjs";
+
+const cache = {};
+
+export default function useFetchData(url) {
+  const [data, setData] = useState([]);
+
+  useEffect(() => {
+    if (!url) return;
+
+    const fetchData = async () => {
+      if (cache[url]) {
+        setData(cache[url]);
+      } else {
+        try {
+          const response = await fetch(url);
+          const resData = await response.json();
+          cache[url] = resData;
+          setData(resData);
+        } catch (err) {
+          captureException(err);
+        }
+      }
+    };
+
+    fetchData();
+  }, [url]);
+
+  return data;
```

Comment:
NIT: тут можно возвращать вот таким образом:
```suggestion
  return [data, isLoading, error.message];
```

Reply:
@Dimirolz Подумай есть ли в проекте необходимость в `loading` (для триггера спинеров) и `error` (например для отображения в уведомлениях админами)